doc/source/dev/missing-bits.rst

```diff
@@ -0,0 +1,260 @@
+
+Code and Documentation Style Guide - The Missing Bits
+=====================================================
+
+This is a collection of coding and documentation guidelines for SciPy that
+are not explicitly stated in the existing guidelines and standards, including
+
+* `PEP-8 <https://www.python.org/dev/peps/pep-0008>`_ Style Guide for Python Code
+* `PEP-257 <https://www.python.org/dev/peps/pep-0257>`_ Docstring Conventions
+* `NumPy docstring standard <https://numpydoc.readthedocs.io/en/latest/format.html>`_
+
+Some of these are trivial, and might not seem worth discussing, but in many
+cases, the issue has come up in a pull request review in either the SciPy
+or NumPy repositories.  If a style issue is important enough that a reviewer
+will require a change before merging, then it is important enough to be
+documented--at least for cases where the issue can be resolved with a simple
+rule.
+
+
+Coding Style and Guidelines
+---------------------------
+
+Required keyword names
+~~~~~~~~~~~~~~~~~~~~~~
+For new functions or methods with more than a few arguments, all parameters
+after the first few "obvious" ones should *require* the use of the keyword
+when given.  This is implemented by including ``*`` at the appropriate point
+in the signature.
+
+For example, a function ``foo`` that operates on a single array but that has
+several optional parameters (say ``method``, ``flag``, ``rtol`` and ``atol``)
+would be defined as::
+
+    def foo(x, *, method='basic', flag=False, rtol=1.5e-8, atol=1-12):
+        ...
+
+To call ``foo``, all parameters other than ``x`` must be given with an
+explicit keyword, e.g. ``foo(arr, rtol=1e-12, method='better')``.
+
+This forces callers to give explicit keyword parameters (which most users
+would probably do anyway even without the use of ``*``), *and* it means
+additional parameters can be added to the function anywhere after the
+``*``; new parameters do not have to be added after the existing parameters.
+
+
+Test functions from `numpy.testing`
```

Comment:
We actually have a link in the doc pointing to NumPy's testing doc. And in NumPy's doc it's says their guide applies to us. We should put the link to NumPy here (https://docs.scipy.org/doc/numpy/reference/testing.html). And in `HACKING.rst.txt` and `development_workflow.rst`, maybe point to here as well.

Reply:
Please note that the new link is https://numpy.org/doc/stable/reference/testing.html

Reply:
The link to the numpy testing guidelines is a few lines down, and thanks to @melissawm, it now has the correct URL.